### PR TITLE
build(deps): bump ibc-go to v3.0.0 and wasmd to v0.27.0

### PR DESCRIPTION
**Pull request by bragaz**
## Description

Superseeds #874
Closes: #872
Closes: #873

Bumps [wasmd](https://github.com/CosmWasm/wasmd) to [v0.27.0](https://github.com/CosmWasm/wasmd/releases/tag/v0.27.0)

Includes also the changes of #874 since the two are strictly dependent.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://docs.cosmos.network/v0.44/building-modules/intro.html)
- [ ] included the necessary unit and integration [tests](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [x] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by RiccardoM — APPROVED** [review]

File: app/app.go [hunk 1]
```diff
@@ -536,7 +544,8 @@ func NewDesmosApp(
 	)
 
 	// Add wasm module route to the ibc router, then set and seal it
-	ibcRouter.AddRoute(wasm.ModuleName, wasm.NewIBCHandler(app.WasmKeeper, app.IBCKeeper.ChannelKeeper))
+	// TODO re-add this in a further PR to update wasmd
```
Comment (dadamu) [inline app/app.go:547]: Shouldn't it be re-added since this PR also includes the wasmd updation?

Comment (RiccardoM) [reply app/app.go]: @dadamu Added back, thanks!

File: app/app.go [hunk 2]
```diff
@@ -531,7 +539,7 @@ func NewDesmosApp(
 		app.BankKeeper,
 		app.StakingKeeper,
 		app.DistrKeeper,
-		app.IBCKeeper.ChannelKeeper,
+		nil, //app.IBCKeeper.ChannelKeeper, TODO re-add this in a further PR to update wasmd
```
Comment (dadamu) [inline app/app.go:542]:
```suggestion
		app.IBCKeeper.ChannelKeeper,
```

Comment (dadamu) [reply app/app.go:542]: Some tests are failing might be because of it.

File: testutil/ibctesting/coordinator.go [hunk 3]
```diff
@@ -35,7 +35,7 @@ func NewCoordinator(t *testing.T, n int) *Coordinator {
 	chains := make(map[string]*TestChain)
 
 	for i := 0; i < n; i++ {
-		chainID := GetChainID(i)
+		chainID := GetChainID(i + 1)
```
Comment (dadamu) [inline testutil/ibctesting/coordinator.go:38]: Why plus 1 here?

Comment (RiccardoM) [reply testutil/ibctesting/coordinator.go:38]: This is just to reflect the following change that has been made inside the standard `Coordinator` implementation with `ibc-go v3.0.0`: 

> (testing) https://github.com/cosmos/ibc-go/pull/774 Added ChainID arg to SetupWithGenesisValSet on the testing app. Coordinator generated ChainIDs now starts at index 1

**Review by dadamu — APPROVED** [review]
Ready to go~

**Pull request merged**